polars/polars-lazy/polars-plan/src/dsl/function_expr/temporal.rs

```diff
@@ -7,43 +7,71 @@ use polars_time::prelude::*;
 use super::*;
 
 #[cfg(feature = "date_offset")]
-pub(super) fn date_offset(s: Series, offset: Duration) -> PolarsResult<Series> {
+pub(super) fn date_offset(s: &[Series]) -> PolarsResult<Series> {
+    let sa = &s[0];
+    let sb = &s[1];
+
     let preserve_sortedness: bool;
-    let out = match s.dtype().clone() {
+    let out = match sa.dtype().clone() {
         DataType::Date => {
-            let s = s
+            let sa = sa
                 .cast(&DataType::Datetime(TimeUnit::Milliseconds, None))
                 .unwrap();
             preserve_sortedness = true;
-            date_offset(s, offset).and_then(|s| s.cast(&DataType::Date))
+            date_offset(&[sa, sb.clone()]).and_then(|s| s.cast(&DataType::Date))
```

Comment:
does it work to just do `date_offset(s)` here and avoid the clone?

Reply:
I did have `&[sa, sb]` but the compiler complained and told me to `.clone()`

I don't think I can use `s` because the `s[0]` needs to be cast to a Datetime:

https://github.com/pola-rs/polars/blob/56fd7849469bf73b14c7cd9057225e1d7649fde1/polars/polars-lazy/polars-plan/src/dsl/function_expr/temporal.rs#L17-L19

(My rust knowledge is limited, so it's possible I'm wrong here.)

Reply:
ah sorry, you're right